covidthailand.py

```diff
@@ -3412,53 +3779,95 @@ def save_plots(df):
+    # fig, ax = plt.subplots(figsize=[20, 10])
+    df['Recovered since 2021-04-01'] = df['2021-04-14':]['Recovered'].cumsum()
+    df['Died since 2021-04-01'] = df['2021-04-01':]['Deaths'].cumsum()
+    df['Cases since 2021-04-01'] = df['2021-04-01':]['Cases'].cumsum()
+    df['Other Active Cases'] = \
+        df['Cases since 2021-04-01'].sub(non_split, fill_value=0).sub(df['Recovered since 2021-04-01'], fill_value=0)
 
     cols = [
-        "Died since 2021-04-01", 
-        "Hospitalized Respirator",
-        "Hospitalized Severe",
-        "Other Active Cases",
-        "Hospitalized Field",
-        "Recovered since 2021-04-01", 
+        'Died since 2021-04-01',
+        'Hospitalized Respirator',
+        'Hospitalized Severe',
+        'Other Active Cases',
+        'Hospitalized Field',
+        'Recovered since 2021-04-01',
     ]
-    # df["2021-04-01":].plot.line(
+    # df['2021-04-01':].plot.line(
     #     ax=ax,
-    #     y="Cases since 2021-04-01",
-    #     #colormap="tab20",
+    #     y='Cases since 2021-04-01',
+    #     #colormap='tab20',
     # )
-    df["2021-04-01":].plot.area(
-        ax=ax,
-        y=cols,
-        colormap="tab20",
-        title='Current outcome of Covid Cases since 1st April 2021\n'
-        f"Updated: {TODAY().date()}\n"        
-        "djay.github.io/covidthailand"
-    )
-    ax.legend([
-#        "Cases since 1st April" 
-        "Deaths from cases since 1st April", 
-        "On Ventilator",
-        "In severe condition",
-        "In Hospital",
-        "In Field Hospital",
-        "Recovered from cases since 1st April",
-    ])
-
-    plt.tight_layout()
-    plt.savefig("outputs/cases_cumulative_3.png")
+#     df['2021-04-01':].plot.area(
+#         ax=ax,
+#         y=cols,
+#         colormap='tab20',
+#         title='Current outcome of Covid Cases since 1st April 2021\n'
+#         f'Updated: {TODAY().date()}\n'
+#         'djay.github.io/covidthailand'
+#     )
+#     ax.legend([
+# #        'Cases since 1st April'
+#         'Deaths from cases since 1st April',
+#         'On Ventilator',
+#         'In severe condition',
+#         'In Hospital',
+#         'In Field Hospital',
+#         'Recovered from cases since 1st April',
+#     ])
+#
+#     plt.tight_layout()
+#     plt.savefig('outputs/cases_cumulative_3.png')
+#     legends = [
+# #        'Cases since 1st April'
+#         'Deaths from cases since 1st April',
+#         'On Ventilator',
+#         'In severe condition',
+#         'In Hospital',
+#         'In Field Hospital',
+#         'Recovered from cases since 1st April'
+#     ]
+    legends = [
+            'Deaths from cases since 1st April',
+            'On Ventilator',
+            'In severe condition',
+            'In Hospital',
+            'In Field Hospital',
+            'Recovered from cases since 1st April'
+    ]
+    plot_area(df=df, png_prefix='cases_cumulative_3', cols_subset=cols,
```

Comment:
This graph is a special one because the data depends on it starting at 1st april. So it can't be worked out for all teh different time periods. Not sure if handle this outside of area_plot, or add more params so it doesn't do all the time periods?

Reply:
Easiest way for that particular plot is to filter the DataFrame before passing it to the `plot_area` function (similar to `plot_area(df=top5.last('30d')` for another plot). Works for you?

Reply:
Same suggested approach: filter the DataFrame so that we pass only relevant data and detect empty DataFrame inside the `plot_area` function via `if df_plot.empty: continue`